pyterrier/index.py

```diff
@@ -269,7 +290,15 @@ class TerrierIndexer:
             "trec.collection.class": "TRECCollection",
     }
 
-    def __init__(self, index_path, *args, blocks=False, overwrite=False, verbose=False, meta_reverse=["docno"], type=IndexingType.CLASSIC, **kwargs):
+    def __init__(self, index_path : str, *args, 
+            blocks : bool = False, 
+            overwrite: bool = False, 
+            verbose : bool = False, 
+            meta_reverse : List[str] = ["docno"],
+            stemmer : Union[None, str, TerrierStemmer] = TerrierStemmer.porter,
+            stopwords : Union[None, TerrierStopwords] = TerrierStopwords.terrier,
+            type=IndexingType.CLASSIC, 
```

Comment:
Can we add tokenizer too?

Reply:
now added. Do I need an enum for this as well?

Reply:
Might be nice for documentation and such

Reply:
ok, added. documentation is improved. note that TerrierIndexer.__init__() is not exposed in terrier-indexing.rst, and the sub-classes pass these arguments in **kwargs. Ways to improve gratefully received.